bin/trait_mapping/get_children_with_keywords.py

```diff
@@ -0,0 +1,56 @@
+import argparse
+
+from cmat.clinvar_xml_io.ontology_uri import OntologyUri
+from cmat.trait_mapping.ols import build_ols_query
+from cmat.trait_mapping.utils import json_request
+
+def append_embedded(results, json_response):
+    if json_response and '_embedded' in json_response:
+        for key in json_response['_embedded']:
+            if key not in results:
+                results[key] = []
+            results[key].extend(json_response['_embedded'][key])
+
+def query_and_depaginate(url):
+    json_response = json_request(url)
+    results = {}
+    append_embedded(results, json_response)
+    while 'next' in json_response['_links']:
+        json_response = json_request(json_response['_links']['next']['href'])
+        append_embedded(results, json_response)
+    return results
+
+def search_in(keywords, text):
+    return set((keyword for keyword in keywords if keyword in text))
+
+def main():
+    parser = argparse.ArgumentParser('Search OLS for children of a term that match certain keywords in their label, description or synonyms')
+    parser.add_argument('--ontology', type=str, default='MONDO', help='Name of the Ontology to find the parent and children')
+    parser.add_argument('--parent_curie', type=str, help='Curie of the parent term', required=True)
+    parser.add_argument('--keywords', type=str, nargs='+', help="Words that must be present in the child's ontology label, description or synonyms to be reported")
+
+    args = parser.parse_args()
+
+
+    keywords = set(args.keywords)
+    db = args.ontology
+    parent_curie = args.parent_curie
+    url = build_ols_query(OntologyUri(parent_curie, db).uri)
+    results = query_and_depaginate(url)
+    for term in results['terms']:
+        if term['ontology_prefix'] == db:
+            results2 = query_and_depaginate(term['_links']['children']['href'])
```

Comment:
Could maybe call this `children_results` or something similar

Reply:
Done